tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -312,23 +389,36 @@ const reducer = createReducer(
       // metadata does not include it.
       const nextCardMetadataMap = {...state.cardMetadataMap};
       const nextCardMetadataList = buildCardMetadataList(nextTagMetadata);
-      const nextCardList = [...state.cardList];
+      const newCardIds = [];
 
       // Create new cards for unseen metadata.
       for (const cardMetadata of nextCardMetadataList) {
         const cardId = getCardId(cardMetadata);
         if (!state.cardMetadataMap.hasOwnProperty(cardId)) {
           nextCardMetadataMap[cardId] = cardMetadata;
-          nextCardList.push(cardId);
+          newCardIds.push(cardId);
         }
       }
 
+      const nextCardList = newCardIds.length
+        ? [...state.cardList, ...newCardIds]
+        : state.cardList;
```

Comment:
can't you spread zero length array? `const nextCardList = [...state.cardList, ...newCardIds]` should suffice?